### ISSUE TITLE
Error in building latest version of SALib

### ISSUE THREAD

**Issue by ConnectedSystems** [conda-forge/salib-feedstock#18]
Issue: Conda build does not successfully complete

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /home/takuya/miniconda3/envs/salib:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
appdirs                   1.4.4                    pypi_0    pypi
attrs                     19.3.0                   pypi_0    pypi
beautifulsoup4            4.9.1            py37hc8dfbb8_0    conda-forge
black                     19.10b0                  pypi_0    pypi
brotlipy                  0.7.0           py37h8f50634_1000    conda-forge
bzip2                     1.0.8                h516909a_2    conda-forge
ca-certificates           2020.6.20            hecda079_0    conda-forge
certifi                   2020.6.20        py37hc8dfbb8_0    conda-forge
cffi                      1.14.0           py37he30daa8_1  
chardet                   3.0.4           py37hc8dfbb8_1006    conda-forge
click                     7.1.2                    pypi_0    pypi
conda                     4.8.3            py37hc8dfbb8_1    conda-forge
conda-build               3.19.2           py37hc8dfbb8_2    conda-forge
conda-package-handling    1.6.0            py37h8f50634_2    conda-forge
cryptography              2.9.2            py37hb09aad4_0    conda-forge
cycler                    0.10.0                   pypi_0    pypi
filelock                  3.0.12             pyh9f0ad1d_0    conda-forge
glob2                     0.7                        py_0    conda-forge
icu                       67.1                 he1b5a44_0    conda-forge
idna                      2.10               pyh9f0ad1d_0    conda-forge
jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
kiwisolver                1.2.0                    pypi_0    pypi
ld_impl_linux-64          2.33.1               h53a641e_7  
libarchive                3.3.3             h3a8160c_1008    conda-forge
libedit                   3.1.20191231         h14c3975_1  
libffi                    3.3                  he6710b0_2  
libgcc-ng                 9.1.0                hdf63c60_0  
libiconv                  1.15              h516909a_1006    conda-forge
liblief                   0.9.0                hf8a498c_1    conda-forge
libstdcxx-ng              9.1.0                hdf63c60_0  
libxml2                   2.9.10               h72b56ed_1    conda-forge
lz4-c                     1.9.2                he1b5a44_1    conda-forge
lzo                       2.10              h14c3975_1000    conda-forge
markupsafe                1.1.1            py37h8f50634_1    conda-forge
matplotlib                3.2.2                    pypi_0    pypi
ncurses                   6.2                  he6710b0_1  
numpy                     1.19.0                   pypi_0    pypi
openssl                   1.1.1g               h516909a_0    conda-forge
pandas                    1.0.5                    pypi_0    pypi
patchelf                  0.11                 he1b5a44_0    conda-forge
pathspec                  0.8.0                    pypi_0    pypi
pip                       20.1.1                   py37_1  
pkginfo                   1.5.0.1                    py_0    conda-forge
psutil                    5.7.2            py37h8f50634_0    conda-forge
py-lief                   0.9.0            py37he1b5a44_1    conda-forge
pycosat                   0.6.3           py37h8f50634_1004    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pyopenssl                 19.1.0                     py_1    conda-forge
pyparsing                 2.4.7                    pypi_0    pypi
pyscaffold                3.2.3                      py_0    conda-forge
pysocks                   1.7.1            py37hc8dfbb8_1    conda-forge
python                    3.7.7                hcff3b4d_5  
python-dateutil           2.8.1                    pypi_0    pypi
python-libarchive-c       2.9                      py37_0    conda-forge
python_abi                3.7                     1_cp37m    conda-forge
pytz                      2020.1             pyh9f0ad1d_0    conda-forge
pyyaml                    5.3.1            py37h8f50634_0    conda-forge
readline                  8.0                  h7b6447c_0  
regex                     2020.7.14                pypi_0    pypi
requests                  2.24.0             pyh9f0ad1d_0    conda-forge
ripgrep                   12.1.1               h516909a_0    conda-forge
ruamel_yaml               0.15.80         py37h8f50634_1001    conda-forge
salib                     1.1.3.post0.dev235+g0bee9ff.dirty           dev_0    <develop>
scipy                     1.5.1                    pypi_0    pypi
setuptools                49.2.0                   py37_0  
six                       1.15.0             pyh9f0ad1d_0    conda-forge
soupsieve                 2.0.1            py37hc8dfbb8_0    conda-forge
sqlite                    3.32.3               h62c20be_0  
tk                        8.6.10               hbc83047_0  
toml                      0.10.1                   pypi_0    pypi
tqdm                      4.47.0             pyh9f0ad1d_0    conda-forge
typed-ast                 1.4.1                    pypi_0    pypi
urllib3                   1.25.9                     py_0    conda-forge
wheel                     0.34.2                   py37_0  
xz                        5.2.5                h7b6447c_0  
yaml                      0.2.5                h516909a_0    conda-forge
zlib                      1.2.11               h7b6447c_3  
zstd                      1.4.4                h6597ccf_3    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```

$ conda info
     active environment : salib
    active env location : /home/takuya/miniconda3/envs/salib
            shell level : 2
       user config file : /home/takuya/.condarc
 populated config files : 
          conda version : 4.8.3
    conda-build version : not installed
         python version : 3.7.7.final.0
       virtual packages : __glibc=2.31
       base environment : /home/takuya/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/takuya/miniconda3/pkgs
                          /home/takuya/.conda/pkgs
       envs directories : /home/takuya/miniconda3/envs
                          /home/takuya/.conda/envs
               platform : linux-64
             user-agent : conda/4.8.3 requests/2.23.0 CPython/3.7.7 Linux/5.4.0-40-generic ubuntu/20.04 glibc/2.31
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```
</details>

Command run  is the usual `conda build`.

The error I eventually encounter comes from the `conda-build` script attempting to run:

```bash
$ python -m pip --disable-pip-version-check wheel --no-deps -w /tmp/tmpvd36fxvr --quiet pyscaffold<3.3a0,>=3.2a0
bash: 3.3a0,: No such file or directory
```
 
If I attempt to run command with the version spec quoted, then it works (no other output at least):

```bash
python -m pip --disable-pip-version-check wheel --no-deps -w /tmp/tmpvd36fxvr --quiet "pyscaffold<3.3a0,>=3.2a0"
``` 

I'm new to conda packaging so I'm currently unsure where this version specification comes from. Modifying `setup.cfg` does not appear to do anything.

**Comment by willu47** [comment]
Hi @ConnectedSystems - thanks for this.  I spend a few hours banging my head against this issue the other day.  The main file to modify is `recipe/meta.yaml`, that's where the install steps for the dependencies are listed.

**Comment by willu47** [comment]
Still [erroring](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=185385&view=logs&j=1bf226d3-0e2f-52d8-fa93-7d9e633347b3&t=ec8d466d-e3b4-5115-4f06-222398f91e6c&l=483) after wrapping the pyscaffold line in quotes 99e8da8f2421a2ef922fbd96e55cba558561cc7a

**Comment by willu47** [comment]
Progress!  Replacing `{{ PYTHON }}` with `python3` in 8f346d8078f66856dcffc23312bbb24a59b1fdd7 allowed the build to complete, but failed at packaging.  So there's something about needed to ensure we install pyscaffold with Python 3, and not the Python 2 environment it seems to be using by default.

**Comment by ConnectedSystems** [comment]
Progress on my end too

Changed the version specification to (note the spaces): `pyscaffold >=3.2a0, <3.3a0`

```
Unsatisfiable dependencies for platform win-64: {"pyscaffold[version='>=3.2a0,<3.3a0']"}
```

Update:

This is weird...

```
Unsatisfiable dependencies for platform win-64: {'pyscaffold'}
```

**Comment by ConnectedSystems** [comment]
Okay, I think I've got it working locally, it's going through the build process now which is the furthest I've ever gotten.

The issue was that `pyscaffold` is not actually available through the default conda channel, so we have to specify conda-forge like so:

`conda build . -c conda-forge`

Will test it on my linux machine too.

EDIT: I see that the build recipe already specifies the conda-forge channel, so I'll dig deeper.

**Comment by ConnectedSystems** [comment]
Last update for the night.

On Windows the build fails with 

```
The process tried to write to a nonexistent pipe.

'3.7' is not recognized as an internal or external command,
operable program or batch file.
```

On my linux machine, it fails with:

```
3.7 -m pip install . --no-deps -vv
/home/takuya/miniconda3/envs/salib/conda-bld/salib_1594985063368/work/conda_build.sh: line 4: 3.7: command not found

subprocess.CalledProcessError: Command '['/bin/bash', '-x', '-o', 'errexit', '/home/takuya/miniconda3/envs/salib/conda-bld/salib_1594985063368/work/conda_build.sh']' returned non-zero exit status 127.
```

My current guess is `{{python}}` resolves to "3.7" for some reason...
